src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/SpatialCoordinateSystemParticipant.cs

```diff
@@ -27,7 +27,7 @@ public class SpatialCoordinateSystemParticipant : DisposableBase, IPeerConnectio
         private bool showDebugVisuals;
         private readonly TaskCompletionSource<ISet<Guid>> peerSupportedLocalizersTaskSource = new TaskCompletionSource<ISet<Guid>>();
 
-        public SocketEndpoint SocketEndpoint { get; }
+        public SocketEndpoint SocketEndpoint { get; set; }
```

Comment:
>public SocketEndpoint SocketEndpoint { get; set; } [](start = 8, length = 50)

Under what circumstances can the endpoint associated with a participant change?

Reply:
we saw locally that Unity buttons ended up having two event calls on a single press based on some input configurations. This caused two socket endpoints to get created for the same ip address. TCPConnectionManager has been updated to prevent creating multiple end points with subsequent event calls, but other networking abstractions may not be as friendly.

Do you believe it is safe to key off ip addresses or can you imagine scenarios where the same ip address creates multiple endpoints for the same shared experience?

Reply:
I could certainly imagine scenarios where this would break, e.g. if we wanted two devices both connected over IPOverUSB to work (on different ports). They would both have 127.0.0.1 as their address but be bound to different ports and have different endpoints.

---
In reply to: [337232321](https://github.com/microsoft/MixedReality-SpectatorView/pull/247#discussion_r337232321) [](ancestors = 337232321)

Reply:
i will revert this now.